File: CommonVB/ByteOps/ByteReader.vb
Imports System.IO

Public Class ByteReader
    Inherits BaseByteReader

    Public Sub New(ByVal Bytes As Byte())
        MyBase.New(Bytes)
    End Sub
    Public Sub New(ByVal nBaseStream As IO.Stream)
        MyBase.New(nBaseStream)
    End Sub
    Public Sub New(ByVal nMem As MemoryStream)
        MyBase.New(nMem)
    End Sub


    Public Function ReadVector3() As Vector3
        Return New Vector3(Me.ReadSingle, Me.ReadSingle, Me.ReadSingle)
    End Function
    Public Function ReadVector2() As Vector2
        Return New Vector2(Me.ReadSingle, Me.ReadSingle)
    End Function
    Public Function ReadVector4() As Vector4
        Return New Vector4(Me.ReadSingle, Me.ReadSingle, Me.ReadSingle, Me.ReadSingle)
    End Function
    Public Function ReadQuaternion() As Quaternion
        Return New Quaternion(Me.ReadSingle, Me.ReadSingle, Me.ReadSingle, Me.ReadSingle)
    End Function
    Public Function ReadMatrix() As Matrix
        Dim M As Matrix
        M.M11 = Me.ReadSingle
        M.M12 = Me.ReadSingle
        M.M13 = Me.ReadSingle
        M.M14 = Me.ReadSingle
        M.M21 = Me.ReadSingle
        M.M22 = Me.ReadSingle
        M.M23 = Me.ReadSingle
        M.M24 = Me.ReadSingle
        M.M31 = Me.ReadSingle
        M.M32 = Me.ReadSingle
        M.M33 = Me.ReadSingle
        M.M34 = Me.ReadSingle
        M.M41 = Me.ReadSingle
        M.M42 = Me.ReadSingle
        M.M43 = Me.ReadSingle
        M.M44 = Me.ReadSingle

        Return M
    End Function



End Class

File: Common.Core/Core/IO/ByteReader.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using Microsoft.Xna.Framework;

namespace MHGameWork.TheWizards.ServerClient
{
    public class TWByteReader : BinaryReader
    {
        public TWByteReader( byte[] Bytes )
            : base( new MemoryStream( Bytes ) )
        {
        }
        public TWByteReader( System.IO.Stream nBaseStream )
            : base( nBaseStream )
        {
        }


        public Vector3 ReadVector3()
        {
            return new Vector3( this.ReadSingle(), this.ReadSingle(), this.ReadSingle() );
        }
        public Vector2 ReadVector2()
        {
            return new Vector2( this.ReadSingle(), this.ReadSingle() );
        }
        public Vector4 ReadVector4()
        {
            return new Vector4( this.ReadSingle(), this.ReadSingle(), this.ReadSingle(), this.ReadSingle() );
        }
        public Quaternion ReadQuaternion()
        {
            return new Quaternion( this.ReadSingle(), this.ReadSingle(), this.ReadSingle(), this.ReadSingle() );
        }
        public Matrix ReadMatrix()
        {
            Matrix M = default( Matrix );
            M.M11 = this.ReadSingle();
            M.M12 = this.ReadSingle();
            M.M13 = this.ReadSingle();
            M.M14 = this.ReadSingle();
            M.M21 = this.ReadSingle();
            M.M22 = this.ReadSingle();
            M.M23 = this.ReadSingle();
            M.M24 = this.ReadSingle();
            M.M31 = this.ReadSingle();
            M.M32 = this.ReadSingle();
            M.M33 = this.ReadSingle();
            M.M34 = this.ReadSingle();
            M.M41 = this.ReadSingle();
            M.M42 = this.ReadSingle();
            M.M43 = this.ReadSingle();
            M.M44 = this.ReadSingle();

            return M;
        }




    }
}
